lib/scout_apm/instant/middleware.rb

```diff
@@ -0,0 +1,116 @@
+module ScoutApm
+  module Instant
+
+    # an abstraction for manipulating the HTML we capture in the middleware
+    class Page
+      def initialize(html)
+        @html = html
+        @to_add_to_head = []
+        @to_add_to_body = []
+      end
+
+      def add_to_head(content)
+        @to_add_to_head << content
+      end
+
+      def add_to_body(content)
+        @to_add_to_body << content
+      end
+
+      def res
+        i = @html.index("</body>")
+        @html = @html.insert(i, @to_add_to_body.join("")) if i
+        i = @html.index("</head>")
+        @html = @html.insert(i, @to_add_to_head.join("")) if i
+        @html
```

Comment:
Is it intentional to silently fail to add stuff to the page if you don't find the indexes?